### DISCUSSION TITLE
How to handle application/x-msgpack response in react

### DISCUSSION THREAD

**Discussion by codekathiravan** [axios/axios#5863] [Q&A]
### Describe the bug

I am using React with typescript. 
.Net 7 api is returning msgpack response
How to read as json ?

### Code snippet

```js
.then(function (response: any) {
                console.log(response,"response");
                var buffer = new Uint8Array(response.data);
                console.log(buffer,"buffer");
                var decoded = decode(buffer);
                console.log(decoded,"decoded");
                handleResponse(response.data);
            })

above code is not working
```

**Comment by jasonsaayman** [comment]
Axios will not decode `application/x-msgpack` as JSON. You need to request the response as binary data and then decode it with a MessagePack library.

In the browser/React, use `responseType: 'arraybuffer'`:

```ts
import { decode } from '@msgpack/msgpack';

const response = await axios.get('/your-url', {
  responseType: 'arraybuffer',
  headers: {
    Accept: 'application/x-msgpack'
  }
});

const decoded = decode(new Uint8Array(response.data));

console.log(decoded);
```

If you use `response.data` without `responseType: 'arraybuffer'`, Axios/browser may treat it as text or try normal response handling, which will not work for MessagePack.

Also make sure your server returns the correct header:

```http
Content-Type: application/x-msgpack
```
